### PR TITLE
docs: Javadocs for remainder of Transfer Manager

### PR DESCRIPTION
**Pull request by sydney-munro**
Javadocs for Upload/Download job, Upload/Download Result, TransferManagerConfig and TransferStatus

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadResult.java [hunk 1]
```diff
@@ -28,6 +28,12 @@
 import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
 import org.checkerframework.checker.nullness.qual.NonNull;
 
+/**
+ * Result for a single download performed by Transfer Manager. This can be for a chunked download or
```
Comment (sydney-munro) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadResult.java:32]: remove chunked

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadJob.java [hunk 2]
```diff
@@ -43,11 +48,23 @@ private DownloadJob(
     this.parallelDownloadConfig = parallelDownloadConfig;
   }
 
+  /**
+   * The list of {@link DownloadResult DownloadResults} for each download request Transfer Manager
+   * for this job. Note calling this method will block the invoking thread until the operation is
```
Comment (sydney-munro) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadJob.java:53]: more verbose

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerConfig.java [hunk 3]
```diff
@@ -142,24 +167,49 @@ private Builder() {
       this.qos = DefaultQos.of();
     }
 
+    /**
+     * Maximum amount of workers to be allocated to perform work in Transfer Manager
+     *
```
Comment (BenWhitehead) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerConfig.java:172]: It's particularly useful to include what the default value is in the javadocs for each of these setter methods.

```suggestion
     * <p><i>Default Value:</i> {@code 2 * }{@link Runtime#getRuntime()}{@code .}{@link
     * Runtime#availableProcessors() availableProcessors()}
     *
```

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferStatus.java [hunk 4]
```diff
@@ -19,14 +19,23 @@
...
+  /** The transfer failed before bytes could be moved. */
   @BetaApi
   FAILED_TO_START,
+  /** The transfer failed after bytes could be moved. */
   @BetaApi
   FAILED_TO_FINISH,
+  /**
+   * The transfer failed because the object/file already exists and skipIfExists was set to true.
+   *
+   * @see ParallelUploadConfig
```
Comment (BenWhitehead) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferStatus.java:34]: We can deep link this
```suggestion
   * @see ParallelUploadConfig.Builder#setSkipIfExists(boolean)
```

**Review by BenWhitehead — APPROVED** [review]

**Pull request merged**